projects/plugins/jetpack/extensions/shared/memberships.js

```diff
@@ -113,7 +113,10 @@ export const setPurchaseResultCookie = function ( premiumContentJWTToken ) {
 	// We will set this in a cookie  - just in case. This will be reloaded in the refresh, when user clicks OK.
 	// But user can close the browser window before clicking OK. IN that case, we want to leave a cookie behind.
 	const hostname = window.location.hostname;
-	const domain = '.' + hostname;
+	let domain = hostname;
+	if ( 'wordpress.com' === hostname ) {
```

Comment:
Not sure about it. I think the hostname should end with `.wordpress.com`, not be equal to `wordpress.com`. Am I right?

Reply:
Yes, you're right. I meant to do a string search, not a match like this. I'll update.

Reply:
String search is prone to errors as well since `'abcwordpress.com'.indexOf( 'wordpress.com' ) !== -1` equals to 'true'. We should use sth like `hostname.endsWith( '.wordpress.com' )` instead.

Reply:
I think this is happening _because_ we are setting the `domain=` to any value. If you try omitting it, it won't add a leading `.` to the cookie domain. I left more details in another comment.